partitura/io/exportmatch.py

```diff
@@ -402,14 +400,18 @@ def matchfile_from_alignment(
     sort_stime = []
     note_lines = []
     
-    # Filter out voice overlap notes using a pd.Dataframe structure
-    spart_df = pd.DataFrame(spart.note_array())
-    voice_overlap_candidates = spart_df.groupby(['onset_beat', 'pitch'])
-    voice_overlap_note_ids = []
-    for _, group in voice_overlap_candidates:
-        if len(group) > 1:
-            voice_overlap_note_ids.extend(np.array(group.values.tolist())[:,-1].tolist())
-            
+    # Get ids of notes which voice overlap
+    spart_array = np.array([list(row) for row in spart.note_array()])
+    onset_pitch_slice = [tuple(i) for i in list(spart_array[:, [0, 6]])]
```

Comment:
There are a lot of for loops in this lines do you think we can do something a bit more efficiently?
I don't understand what line 404 does (transforming the note array which is a structured numpy array to a numpy array, and then you are still looping over again with a for loop to take the onsets. While this can be easily done by:
```suggestion
   onset_pitch_slice = spart.note_array()[["onset_beat", "pitch"]]
```
But also lines 408-413 are each one a loop, is it possible to condense it to fewer?